cli/cmd/gateway/list/command.go

```diff
@@ -0,0 +1,312 @@
+package read
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	helmcli "helm.sh/helm/v3/pkg/cli"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/hashicorp/consul-k8s/cli/common"
+	"github.com/hashicorp/consul-k8s/cli/common/flag"
+	"github.com/hashicorp/consul-k8s/cli/common/terminal"
+)
+
+type Command struct {
+	*common.BaseCommand
+
+	kubernetes client.Client
+	restConfig *rest.Config
+
+	set *flag.Sets
+
+	flagAllNamespaces    bool
+	flagGatewayKind      string
+	flagGatewayNamespace string
+	flagKubeConfig       string
+	flagKubeContext      string
+	flagOutput           string
+
+	initOnce sync.Once
+	help     string
+}
+
+func (c *Command) Help() string {
+	c.initOnce.Do(c.init)
+	return fmt.Sprintf("%s\n\nUsage: consul-k8s gateway list [flags]\n\n%s", c.Synopsis(), c.help)
+}
+
+func (c *Command) Synopsis() string {
+	return "Inspect the configuration for all Gateways in the Kubernetes cluster or a given Kubernetes namespace."
+}
+
+// init establishes the flags for Command
+func (c *Command) init() {
+	c.set = flag.NewSets()
+
+	f := c.set.NewSet("Command Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "namespace",
+		Target:  &c.flagGatewayNamespace,
+		Usage:   "The Kubernetes namespace to list Gateways in.",
+		Aliases: []string{"n"},
+	})
+	f.BoolVar(&flag.BoolVar{
+		Name:    "all-namespaces",
+		Target:  &c.flagAllNamespaces,
+		Default: false,
+		Usage:   "List Gateways in all Kubernetes namespaces.",
+		Aliases: []string{"A"},
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "output",
+		Target:  &c.flagOutput,
+		Usage:   "Output the Gateway configuration as 'json' in the terminal or 'archive' as a zip archive named 'gateways.zip' in the current directory.",
+		Default: "archive",
+		Aliases: []string{"o"},
+	})
+
+	f = c.set.NewSet("Global Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "kubeconfig",
+		Aliases: []string{"c"},
+		Target:  &c.flagKubeConfig,
+		Usage:   "Set the path to a kubeconfig file.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:   "context",
+		Target: &c.flagKubeContext,
+		Usage:  "Set the Kubernetes context to use.",
```

Comment:
Should there be Aliases for this flag?

Reply:
I just made it consistent with the existing `consul-k8s proxy list` command which [doesn't specify an alias](https://github.com/hashicorp/consul-k8s/blob/acb1af85a012b3600480f0120a9d772cd4ad304f/cli/cmd/proxy/list/command.go#L89-L94). I'm not finding any other commands that specify an alias, so I think it makes sense not to have one here for consistency.